docs/proposal.md

```diff
@@ -0,0 +1,108 @@
+## Background
+Containerd is a core container runtime, which provides the minimum set of functionalities to manage the complete container lifecycle of its host system, including container execution and supervision, image distribution and storage, etc.
+
+Containerd was previously part of Docker, [introduced in Docker 1.11](https://blog.docker.com/2016/04/docker-engine-1-11-runc/), used to manage [runC](https://runc.io/) containers on the node. As is shown below, it creates a containerd-shim for each container, and the shim manages the lifecycle of its corresponding container.
+![containerd](./containerd.png)
+
+In Dec. 2016, Docker spun it out into a standalone component, and donated it to [CNCF](https://www.cncf.io/) in Mar. 2017.
+
+## Motivation
+Containerd is a potential alternative to Docker as the runtime for Kubernetes cluster. *Compared with Docker*, containerd has pros and cons.
+### Pros
+* **Stability**: Containerd has limited scope and slower feature velocity, which is expected to be more stable.
+* **Compatibility**: The scope of containerd meets Kubernetes' requirement pretty well.
+* **Performance**:
+  * Containerd consumes less resource because it's only a subset of Docker;
+  * Containerd CRI integration eliminates an extra hop in the stack (as is shown below). ![performance](./performance.png)
+* **Neutral Foundation**: Containerd is part of CNCF now.
+### Cons
+* **User Adoption**:
+  * Ideally, Kubernetes users don't interact with the underlying container runtime directly. However, for the lack of debug toolkits, sometimes users still need to login the node to debug with Docker CLI directly.
+  * Containerd also provides barebone CLIs [ctr](https://github.com/containerd/containerd/tree/master/cmd/ctr) and [dist](https://github.com/containerd/containerd/tree/master/cmd/dist) for development and debugging purpose, but it needs time to educate users.
+* **Maturity**: The rescoped containerd is pretty new, and it's still under heavy development.
+## Goals
+* Make sure containerd constantly meet the requirement of Kubernetes.
+* Implement containerd CRI shim and make sure it provides equivalent functionality, usability and debuggability.
+* Improve Kubernetes by taking advantage of the flexibility provided by containerd.
+## Design
+The following sections discuss the design aspects of the containerd CRI integration. For the purposes of this doc, the containerd CRI integration will be referred to as `CRI-containerd`.
+### Container Lifecycle
+CRI-containerd relies on containerd to manage container lifecycle.
+
+Ideally, CRI-containerd only needs to do api translation and information reorganization. However, CRI-containerd needs to maintain some metadata because:
+* There is a mismatch between container state machine of containerd and CRI - containerd doesn't support `Stop`, so CRI-containerd needs to maintain the `Stopped` state.
```

Comment:
Then we should PR their readme...  which states: 
```
Create,start, stop pause, resume exec, signal, delete
```
see https://github.com/containerd/containerd